### PR TITLE
Net 6820 customize mesh gateway limits

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
Allow users of Mesh Gateway to add custom circuit breakers to the mesh gateway XDS config by using a ServiceDefaults object named MeshGateway

- We based this implementation on the first option described in this comment since we didn't want to blow out the scope of this story by adding a new config object, even though we believe that to be the better long term strategy. https://github.com/hashicorp/consul/issues/13589#issuecomment-1310873473

### Testing & Reproduction steps
- Using a built image, tested creating a mesh gateway to make sure the snapshot was updated appropriately 
- Added a new golden file test to cover adding XDS

### PR Checklist

* [ X ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: .changelog/20945.txt
Comment (sarahalsmiller) [inline .changelog/20945.txt:2]: Would like some help phrasing this better

File: agent/proxycfg/snapshot.go
Comment (sarahalsmiller) [reply agent/proxycfg/snapshot.go:298]: Accident. Deleting it.

File: agent/proxycfg/testing_mesh_gateway.go
Comment (jm96441n) [inline agent/proxycfg/testing_mesh_gateway.go:16]: nit but these probably should be in their own block

File: agent/proxycfg/testing_mesh_gateway.go
Comment (jm96441n) [inline agent/proxycfg/testing_mesh_gateway.go:19]: can we write/use an existing `toPointer` method rather than pull in this dep for this? could be something like

```
func pointerTo[T any] (v T) *T {
   return &v
}
```

File: agent/proxycfg/testing_mesh_gateway.go
Comment (jm96441n) [inline agent/proxycfg/testing_mesh_gateway.go:269]: should this comment be here?

File: agent/proxycfg/mesh_gateway.go
Comment (NiniOak) [inline agent/proxycfg/mesh_gateway.go:130]: would be nice to have a comment here to say what this block of code is doing similar to the ones above

Comment (sarahalsmiller) [reply agent/proxycfg/mesh_gateway.go:130]:
```suggestion
// Watch for service default object that matches this mesh gateway's name
```

**Review by NiniOak — APPROVED** [review]
I think this looks good so I'm going to approve. You can wait for approval from others who have more context on this feature.

File: agent/proxycfg/mesh_gateway.go
Comment (nathancoleman) [inline agent/proxycfg/mesh_gateway.go:676]: It looks like `UpstreamConfig` and `UpstreamConfig.Defaults` are possibly `nil` here

**Review by nathancoleman — CHANGES_REQUESTED** [review]
LGTM otherwise

**Pull request merged**
